packages/auth/src/storage/indexeddb.js

```diff
@@ -416,43 +455,42 @@ fireauth.storage.IndexedDB.prototype.onIDBRequest_ =
  * @override
  */
 fireauth.storage.IndexedDB.prototype.set = function(key, value) {
-  var isLocked = false;
-  var dbTemp;
-  var self = this;
-  return this.initializeDbAndRun_()
-      .then(function(db) {
-        dbTemp = db;
-        var objectStore = self.getDataObjectStore_(
-            self.getTransaction_(dbTemp, true));
-        return self.onIDBRequest_(objectStore.get(key));
-      })
-      .then(function(data) {
-        var objectStore = self.getDataObjectStore_(
-            self.getTransaction_(dbTemp, true));
+  let isLocked = false;
+  return this
+    .withRetry_((db) => {
```

Comment:
Does closure compile it to regular functions? I wasn't sure if you can use arrow functions in closure.

Reply:
seems to compile to `function() {}.bind(this)` https://github.com/google/closure-compiler/issues/2853